File: diploma-server/src/test/java/org/arhor/diploma/data/persistence/converter/BitSet64ConverterTest.java
package org.arhor.diploma.data.persistence.converter;

import org.arhor.diploma.commons.BitSet64;
import org.arhor.diploma.testutils.RandomParameter;
import org.junit.jupiter.api.AfterAll;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

import static org.assertj.core.api.Assertions.assertThat;

@ExtendWith({RandomParameter.Resolver.class})
class BitSet64ConverterTest {

    private static BitSet64Converter converterUnderTest;

    @BeforeAll
    static void setUp() {
        converterUnderTest = new BitSet64Converter();
    }

    @AfterAll
    static void tearDown() {
        converterUnderTest = null;
    }

    @Test
    void shouldCorrectlySerializeAndDeserializeValue(@RandomParameter final long valueBefore) {
        // when
        BitSet64 bitSet64 = converterUnderTest.convertToEntityAttribute(valueBefore);
        Long valueAfter = converterUnderTest.convertToDatabaseColumn(bitSet64);

        // then
        assertThat(bitSet64).isNotNull();
        assertThat(valueAfter).isEqualTo(valueBefore);
    }
}

File: app-server/module-main/src/test/kotlin/com/github/arhor/linden/dragon/tavern/data/persistence/BitSet64ConverterTest.kt
package com.github.arhor.linden.dragon.tavern.data.persistence

import com.github.arhor.linden.dragon.tavern.data.persistence.converter.BitSet64ReadingConverter
import com.github.arhor.linden.dragon.tavern.data.persistence.converter.BitSet64WritingConverter
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeAll
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith

@ExtendWith(com.github.arhor.linden.dragon.tavern.testutils.RandomParameter.Resolver::class)
internal class BitSet64ConverterTest {

    @Test
    fun `should correctly serialize and deserialize value`(
        // given
        @com.github.arhor.linden.dragon.tavern.testutils.RandomParameter valueBefore: Long
    ) {
        // when
        val bitSet64 = readingConverter.convert(valueBefore)
        val valueAfter = writingConverter.convert(bitSet64)

        // then
        assertThat(bitSet64).isNotNull
        assertThat(valueAfter).isEqualTo(valueBefore)
    }

    companion object {
        private lateinit var readingConverter: BitSet64ReadingConverter
        private lateinit var writingConverter: BitSet64WritingConverter

        @BeforeAll
        @JvmStatic
        fun setUp() {
            readingConverter = BitSet64ReadingConverter()
            writingConverter = BitSet64WritingConverter()
        }
    }
}
